File: start.kt
fun LogisticInfo(products: List<Product>) {
  print(products)
}

fun main() {
  val products = listOf(
    Product(
      id = "1234",
      name = "Camisa polo",
      slas = arrayOf("Entrega Agendada", "Entrega Correios")
    ),
    Product(
      id = "567",
      name = "Calça jeans",
      slas = arrayOf("Entrega Correios", "Retirada em loja")
    ),
    Product(
      id = "890",
      name = "Vestido",
      slas = arrayOf("Retirada em loja")
    )
  )
  LogisticInfo(products)
}

data class Product(val id: String, val name: String, val slas: Array<String>)

File: start.go
package main
import "fmt"

type product struct{
  id int
  name string
  sla []string
}

func logistic(products []product) {
  fmt.Println(products)
}

func main() {
  polo := product{1234, "Camisa Polo", []string{"Entrega Agendada", "Entrega Correios"}}
  jeans := product{567, "Calça jeans", []string{"Entrega Correios", "Retirada em loja"}}
  dress := product{890, "Vestido", []string{"Retirada em loja"}}
  products := []product{polo, jeans, dress}
  logistic(products)
}
